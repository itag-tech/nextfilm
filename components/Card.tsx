import { FC } from "react"
import clsx from "clsx"
import Image from "next/image"

import { Movie } from "../models/Movie"
import placeholderImage from "../public/placeholder-film.png"

type CardProps = {
  movie: Movie
}

const Card: FC<CardProps> = ({ movie }) => {

  const poster = (movie.Poster === "N/A" && placeholderImage) || movie.Poster

  return (
    <div className={clsx("flex flex-col justify-center items-center bg-white bg-opacity-50 rounded-lg shadow-md truncate h-64 w-80")}>
      <Image
        src={poster}
        alt={movie.Title}
        width={200}
        height={200}
        className={clsx("h-32 w-32 object-cover shadow-lg")}
      />
      <div className={clsx("flex flex-col justify-center items-center")}>
        <p className={clsx("text-xl font-medium truncate max-w-xs text-purple-700 m-4 px-6")}>{movie.Title}</p>
        <p className={clsx("text-sm font-medium text-purple-400")}>{movie.Year}</p>
      </div>
    </div >
  )
}

export default Card