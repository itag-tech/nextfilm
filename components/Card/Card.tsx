import { FC } from "react"
import clsx from "clsx"
import Image from "next/image"
import { getPoster } from "../../utils/movieUtils"
import { Movie } from "../../models/Movie"

type CardProps = {
  movie: Movie
}

const Card: FC<CardProps> = ({ movie }) => {
  return (
    <div
      className={clsx("flex flex-col justify-center items-center bg-white",
        "bg-opacity-50 rounded-lg shadow-md truncate w-full p-6")}
    >
      <Image
        src={getPoster(movie)}
        alt={movie.Title}
        width={200}
        height={200}
        className={clsx("h-32 w-32 object-cover shadow-lg")}
      />
      <div className={clsx("flex flex-col justify-center items-center")}>
        <p className={clsx("text-sm md:text-lg font-medium truncate max-w-xxs xl:max-w-xs text-purple-700 m-4 px-6")}>{movie.Title}</p>
        <p className={clsx("text-sm font-medium text-purple-400")}>{movie.Year}</p>
      </div>
    </div >
  )
}

export default Card