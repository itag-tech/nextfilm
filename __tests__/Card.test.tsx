import { describe, expect, it } from "vitest"
import { render, screen, within } from '@testing-library/react'
import Card from "../components/Card"

describe("Card", () => {
  it("should render correctly", () => {
    const movie = {
      Title: "The Lord of the Rings: The Fellowship of the Ring",
      Year: "2001",
      imdbID: "tt0120737",
      Type: "movie",
      Poster: "https://m.media-amazon.com/images/M/MV5BMTMwNjQ5NjQxN15BMl5BanBnXkFtZTgwNjQ2NjUyMjE@._V1_SX300.jpg",
      Plot: "A meek Hobbit from the Shire and eight companions set out on a journey to destroy the powerful One Ring and save Middle-earth from the Dark Lord Sauron.",
      Gender: "Action, Adventure, Drama",
    }
    const { container } = render(<Card movie={movie} />)
    expect(screen.getByRole('img', { name: movie.Title })).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})