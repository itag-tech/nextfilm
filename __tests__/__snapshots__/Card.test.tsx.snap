// Vitest Snapshot v1

exports[`Card > should render correctly 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center bg-white bg-opacity-50 rounded-lg shadow-md truncate h-64 w-80"
  >
    <img
      alt="The Lord of the Rings: The Fellowship of the Ring"
      class="h-32 w-32 object-cover shadow-lg"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FM%2FMV5BMTMwNjQ5NjQxN15BMl5BanBnXkFtZTgwNjQ2NjUyMjE%40._V1_SX300.jpg&w=640&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FM%2FMV5BMTMwNjQ5NjQxN15BMl5BanBnXkFtZTgwNjQ2NjUyMjE%40._V1_SX300.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fm.media-amazon.com%2Fimages%2FM%2FMV5BMTMwNjQ5NjQxN15BMl5BanBnXkFtZTgwNjQ2NjUyMjE%40._V1_SX300.jpg&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
    <div
      class="flex flex-col justify-center items-center"
    >
      <p
        class="text-xl font-medium truncate max-w-xs text-purple-700 m-4 px-6"
      >
        The Lord of the Rings: The Fellowship of the Ring
      </p>
      <p
        class="text-sm font-medium text-purple-400"
      >
        2001
      </p>
    </div>
  </div>
</div>
`;
